src/system.cpp

```diff
@@ -0,0 +1,57 @@
+#include "system.hpp"
+
+#include "location_service.h"
+#include "Particle.h"
+#include "cli/conio.hpp"
+#include "consts.hpp"
+
+char SYS_deviceID[32];
+
+SystemDesc_t systemDesc, *pSystemDesc = &systemDesc;
+SystemFlags_t systemFlags;
+
+
+static int SYS_initGPS();
+
+
+int SYS_initSys(void)
+{
+    memset(pSystemDesc, 0, sizeof(SystemDesc_t));
+    systemDesc.deviceID = SYS_deviceID;
+    systemDesc.flags = &systemFlags;
+
+    memset(SYS_deviceID, 0, 32);
+    strncpy(SYS_deviceID, System.deviceID(), 31);
```

Comment:
`System.deviceID` returns a String object, not a cstr - we should not naively `strncpy` it!